tasklist/client/src/modules/utils/formatDateRelative.ts

```diff
@@ -45,44 +46,64 @@ type Resolution =
       type: 'week' | 'months' | 'years';
     };
 
-const TEXT_DATE_LANG_EN: TextMap = {
-  NOW: "'Now'",
-  IN_ONE_MINUTE: "'In 1 minute'",
-  IN_X_MINUTES: "'In {amount} minutes'",
-  ONE_MINUTE_AGO: "'1 minute ago'",
-  X_MINUTES_AGO: "'{amount} minutes ago'",
-  TODAY: "'Today'",
-  YESTERDAY: "'Yesterday'",
-  TOMORROW: "'Tomorrow'",
-  WEEK_FMT: 'EEEE',
-  MONTH_FMT: 'd MMM',
-  YEARS_FMT: 'd MMM yyyy',
-};
+const getConfiguredDateKey = (
+  key: TextMapKeys,
+  isSpeech: boolean,
+  includeTime: boolean,
+) => {
+  let keySuffix = '';
 
-const SPEECH_DATE_LANG_EN: TextMap = {
-  ...TEXT_DATE_LANG_EN,
-  MONTH_FMT: "do 'of' MMMM",
-  YEARS_FMT: "do 'of' MMMM, yyyy",
-};
+  if (includeTime) {
+    keySuffix += '_withTime';
+  }
 
-const TEXT_DATETIME_LANG_EN: TextMap = {
-  ...TEXT_DATE_LANG_EN,
-  TODAY: TEXT_DATE_LANG_EN['TODAY'] + ', HH:mm',
-  YESTERDAY: TEXT_DATE_LANG_EN['YESTERDAY'] + ', HH:mm',
-  TOMORROW: TEXT_DATE_LANG_EN['TOMORROW'] + ', HH:mm',
-  WEEK_FMT: 'EEEE, HH:mm',
-  MONTH_FMT: 'd MMM, HH:mm',
-  YEARS_FMT: 'd MMM yyyy, HH:mm',
+  if (isSpeech) {
+    keySuffix += '_speech';
+  }
+
+  return `relativeDateFormat_${key}${keySuffix}`;
 };
 
-const SPEECH_DATETIME_LANG_EN: TextMap = {
-  ...TEXT_DATE_LANG_EN,
-  TODAY: TEXT_DATE_LANG_EN['TODAY'] + " 'at' HH:mm",
-  YESTERDAY: TEXT_DATE_LANG_EN['YESTERDAY'] + " 'at' HH:mm",
-  TOMORROW: TEXT_DATE_LANG_EN['TOMORROW'] + " 'at' HH:mm",
-  WEEK_FMT: "EEEE 'at' HH:mm",
-  MONTH_FMT: "do 'of' MMMM 'at' HH:mm",
-  YEARS_FMT: "do 'of' MMMM, yyyy 'at' HH:mm",
+const getConfiguredTranslatedFormat = (
+  key: TextMapKeys,
+  isSpeech: boolean,
+  includeTime: boolean,
+  options = {},
+) => {
+  let value = '';
+
+  // Allows inheritance of the format if the time or speech equivalent is not defined
+  if (isSpeech && includeTime) {
+    value = t(getConfiguredDateKey(key, isSpeech, includeTime), {
+      ...options,
+      defaultValue: '',
+    });
+  }
+  if (includeTime) {
+    value =
+      value ||
+      t(getConfiguredDateKey(key, false, includeTime), {
+        ...options,
+        defaultValue: '',
+      });
+  }
+  if (isSpeech) {
+    value =
+      value ||
+      t(getConfiguredDateKey(key, isSpeech, false), {
+        ...options,
+        defaultValue: '',
+      });
+  }
+  value =
+    value ||
+    t(getConfiguredDateKey(key, false, false), {...options, defaultValue: ''});
+
+  if (typeof value !== 'string') {
+    console.warn(`Unexpected translation type for 'relativeDateFormat_${key}'`);
+  }
+
+  return value;
```

Comment:
```suggestion
  if (isSpeech && includeTime) {
    return  t(getConfiguredDateKey(key, isSpeech, includeTime), {
      ...options,
      defaultValue: '',
    });
  }
  if (includeTime) {
    return t(getConfiguredDateKey(key, false, includeTime), {
        ...options,
        defaultValue: '',
      });
  }
  if (isSpeech) {
    return t(getConfiguredDateKey(key, isSpeech, false), {
        ...options,
        defaultValue: '',
      });
  }
  
  return t(getConfiguredDateKey(key, false, false), {...options, defaultValue: ''});
```

Reply:
Can't return directly or it wouldn't "Inherit" the simpler keys. But I will simplify this a little.